### DISCUSSION TITLE
Is there constexpr built-in for literal types constructor?

### DISCUSSION THREAD

**Discussion by Xyncgas** [nlohmann/json#2371] [Q&A] [answered]
Since we can do consst expressions like :
constexpr int a = 0;
with native types,

can we do
constexpr json a = "{\"a\":\"b\"}"; 

because I find it to be useful, a thought that this might be an interesting toy to play with. I am getting error that there's no constexpr constructor so maybe that's my clues, and I kinda have to give up since I am not really good at programming yet, mostly because I am lazy to dig into the library and so wanna check if it's already built-in or something, so would be interesting if it's something people had brought up before

**Comment by nlohmann** [comment, marked as answer]
No, this is not possible. Internally, we rely on `std::vector` and `std::map`, and these containers are not constexpr.

**Reply by gregmarr** [reply]
There is `constexpr` support in `std::vector` with C++20, but not `std::map`, so it would be possible for a subset of the library to be `constexpr`, but it's unlikely that this would be a usable subset as it eliminates the object type.  You could only have arrays of non-object values.

**Comment by Xyncgas** [comment]
well since a literal is all the information we needed to build a json object, instead of doing it at run time, it is attempting that I feel like using a static std::array for this one use case is possible, and if you are using template and stuff, then perhaps it doesnt need to change that much for .size() and .empty() stuff like that to still works.

**Reply by Xyncgas** [reply]
and I am sorry if it's too much to ask

**Reply by nlohmann** [reply]
This would mean a complete change of everything this library is doing. Maybe this is what you're looking for: https://www.youtube.com/watch?v=PJwd4JLYJJY

**Reply by pasbi** [reply]
Does that mean we get `constexpr nlohmann::json` once `std::vector` and `std::map` become `constexpr`?
